webpack/ForemanInventoryUpload/Components/PageHeader/__tests__/PageHeader.test.js

```
@@ -1,13 +1,107 @@
+import React from 'react';
+import { Provider } from 'react-redux';
+import configureMockStore from 'redux-mock-store';
+import thunk from 'redux-thunk';
+import { render, screen } from '@testing-library/react';
 import { testComponentSnapshotsWithFixtures } from '@theforeman/test';
 
 import PageHeader from '../PageHeader';
 
+// Mock the useAdvisorEngineConfig hook
+jest.mock('../../../../common/Hooks/ConfigHooks', () => ({
+  useAdvisorEngineConfig: jest.fn(),
+}));
+
+import { useAdvisorEngineConfig } from '../../../../common/Hooks/ConfigHooks';
+
+const middlewares = [thunk];
+const mockStore = configureMockStore(middlewares);
+
+// Mock store configuration with required data for PageHeader components
+const renderOptions = {
+  API: {
+    INVENTORY_SETTINGS: {
+      response: { 
+        subscriptionConnectionEnabled: true,
+        insightsMinimalDataCollection: false 
+      },
+    },
+  },
+  ForemanRhCloud: {
+    inventoryUpload: {
+      accountsList: {
+        CloudConnectorStatus: {},
+      },
+    },
+  },
+};
+
 const fixtures = {
   'render without Props': {},
   /** fixtures, props for the component */
 };
 
 describe('PageHeader', () => {
+  beforeEach(() => {
+    // Reset mock before each test
+    useAdvisorEngineConfig.mockReset();
```

You can just add the property to the test store, and then you won't have to mock the accessor

---

I’m not sure I fully understand what you mean. Could you clarify?

---

I tried your suggestion to add the property directly to the test store instead of mocking the useAdvisorEngineConfig hook, but it failed due to complex interactions between the hook's dependency on useAPI and the existing Jest mocking infrastructure. The useAPI hook wasn't properly reading from our Redux store because of conflicting mocks in the test environment. Given these technical challenges, I think the current approach with mockUseAdvisorEngineConfig.mockReturnValue() is actually more appropriate here - it's simpler, and avoids the fragile API mocking chain. The hook-level mock effectively tests the same logic (component renders correctly based on IoP mode) without the infrastructure complexity.

---

I was able to replace the direct hook mocking with a store-based testing approach for `use_iop_mode`. The issue was that `useAdvisorEngineConfig` uses `useAPI` from `foremanReact`, which has a global mock that ignores our test store state. My solution: Mock `useAPI` directly and created a shared helper `setupAdvisorEngineTest(mockUseAPI, true/false)` that configures both the mock and store state consistently.